### DISCUSSION TITLE
Realtime Event Triggers

### DISCUSSION THREAD

**Discussion by niklasarnitz** [pocketbase/pocketbase#1169] [Q&A]
Hello, I would like to inquire about a rather specific use-case of the permission system and the realtime subscription event system.

For example, let's view the following database schema:
Table 1: Users:
- FooBar
- BarFoo

Table 2: Items:
- Foooooooo, User Relation -> BarFoo
- Barrrrrrrrrr, User Relation -> Foobar

In the Items Table, the access control is set in the following way: `user = @request.auth.id` (Where user is the user Relation in the items table).

If i now subscribe to real time changes using the JS SDK, the question would be the following:
```typescript
client.collection('Items').subscribe('*', (data) => {
     fetchSingleItem(data.id);
});
```
Do the subscription change events only trigger on changes made to items the user can actually see or do they happen when any value in the table is changed? If the latter is the case, my example implementation would not work, because it would error if another user changed an item the current user does not have access to.

It would be nice, if you could clarify, whats the case here.

**Comment by ganigeorgiev** [comment]
I'm sorry, I don't understand your example.

> Do the subscription change events only trigger on changes made to items the user can actually see or do they happen when any value in the table is changed?

Any change done to the related record model **through the internal `app.Dao()` instance** (eg. from an API request) will trigger the realtime event. The subscribed client will receive the event only if they have "List API rule" access (in case the wildcard topic is used) or "View API rule" access (in case the subscription is for a single record id).

To further clarify, manual/direct changes in the SQLite database will not trigger the realtime events.

**Reply by niklasarnitz** [reply]
Okay, this was the answer I looking for.
That the realtime events are only triggered, when there are sufficient permissions to access the changed item.
